aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2464,10 +2464,19 @@ static Tensor& linalg_norm_out_impl(Tensor& result, const Tensor& self, const op
       "dtype = ", dtype, ", out.dtype = ", result.scalar_type());
   }
   int64_t ndim = self.dim();
+  std::vector<int64_t> dim_ = opt_dim.has_value() ? opt_dim.value().vec() : make_dim_list(ndim);
+
+  bool str_ord_non_matrix_norm = false;
+
   if (opt_str_ord.has_value()) {
-    // 'ord' is string
     auto str_ord = opt_str_ord.value();
     check_str_ord_valid(str_ord, opt_dim, ndim);
+    str_ord_non_matrix_norm = (str_ord != "fro") && (dim_.size() != 2 || dim_.size() != 1);
```

Comment:
What does this `bool` flag mean? It's very difficult to read. It might be better to negate the meaning of the variable and call it `vector_norm_or_frob = (str_ord == "fro") || (dim_.size() == 2) || (dim_.size() == 1);`. Then dispatch to `linalg_vector_norm` in this case. The next question would be, why do we need `frobenius_norm_out` at all? That's just the usual vector norm.

Could you investigate whether this function has any benefits over `at::linalg_vector_norm` at all?

In fact, while looking at this, I found https://github.com/pytorch/pytorch/issues/52712. I think with little effort you could fix that issue as well as it's very relevant here.